File: non_native/lib/view/components/bird_list_view.dart
import 'package:flutter/material.dart';

import '../../model/bird.dart';
import 'bird_card.dart';
import 'delete_bird_dialog.dart';
import 'loading_dialog.dart';

class BirdListView extends StatefulWidget {
  final List<Bird> birds;
  final bool isLoading;
  final Function(int) onDeleteBird;
  final Function(int) onEditBird;

  const BirdListView({
    super.key,
    required this.birds,
    required this.isLoading,
    required this.onDeleteBird,
    required this.onEditBird,
  });

  @override
  State<BirdListView> createState() => _BirdListViewState();
}

class _BirdListViewState extends State<BirdListView> {
  bool showDialog = false;
  Bird? birdToDelete;

  @override
  Widget build(BuildContext context) {
    return Stack(
      children: [
        if (widget.isLoading) const LoadingDialog(),

        Padding(
          padding: const EdgeInsets.symmetric(horizontal: 15.0),
          child: MediaQuery.removePadding(
            context: context,
            removeTop: true,
            child: ListView.separated(
              itemCount: widget.birds.length,
              itemBuilder: (context, index) {
                final bird = widget.birds[index];
                return BirdCard(
                  name: bird.name,
                  order: bird.order,
                  family: bird.family,
                  habitatTemp: bird.habitat.name,
                  number: bird.sightCount,
                  onEditClick: () => widget.onEditBird(bird.id),
                  onDeleteClick: () {
                    setState(() {
                      birdToDelete = bird;
                      showDialog = true;
                    });
                  },
                );
              },
              separatorBuilder: (context, index) {
                return const SizedBox(height: 30);
              },
            ),
          ),
        ),

        //confirmation dialog
        if (showDialog)
          DeleteBirdDialog(
            showDialog: showDialog,
            birdToDelete: birdToDelete,
            onDeleteConfirmed: (id) {
              widget.onDeleteBird(id);
              setState(() {
                showDialog = false;
              });
            },
            onDismiss: () {
              setState(() {
                showDialog = false;
              });
            },
          ),
      ],
    );
  }
}

File: native_upgrade/app/src/main/java/com/example/myapplication/ui/components/BirdListView.kt
package com.example.app.ui.components

import android.util.Log
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.example.app.birds.domain.model.Bird
import com.example.app.ui.viewmodels.BirdsViewState

@Composable
fun BirdListView(
    modifier: Modifier = Modifier,
    state: BirdsViewState,
    onDeleteBird: (Int) -> Unit,
    onEditBird: (Int) -> Unit
) {
    LoadingDialog(isLoading = state.isLoading)

    var showDialog by remember { mutableStateOf(false) }
    var birdToDelete by remember { mutableStateOf<Bird?>(null) }

    Box(modifier = modifier.fillMaxSize()) {
        LazyColumn(
            modifier
                .fillMaxSize()
                .padding(horizontal = 15.dp),
            verticalArrangement = Arrangement.spacedBy(30.dp)
        ) {
            items(state.birds) { bird ->
                BirdCard(
                    name = bird.name,
                    order = bird.order,
                    family = bird.family,
                    habitatTemp = bird.habitat.name,
                    number = bird.sightCount,
                    onEditClick = { onEditBird(bird.id) },
                    onDeleteClick = {
                        Log.d("Test", "Delete clicked for ${bird.name}")
                        birdToDelete = bird
                        showDialog = true
                    })
            }
        }

        DeleteBirdDialog(
            showDialog = showDialog,
            birdToDelete = birdToDelete,
            onDeleteConfirmed = { id ->
                onDeleteBird(id)
                showDialog = false
            },
            onDismiss = {
                showDialog = false
            }
        )
    }


}